src/adapters/DiscordLoggerAdapter.ts

```
@@ -0,0 +1,67 @@
+/* eslint-disable no-console */
+import { LoggerAdapter, LoggerAdapterConfig, LogMessage } from './LoggerAdapter';
+import HttpAdapterFactory, { HttpAdapter } from '@luckbox/http-adapter-factory';
+
+type Config = LoggerAdapterConfig & {
+  webhookId: string,
+  token: string,
+  regex: RegExp,
+}
+
+export default class DiscordLoggerAdapter extends LoggerAdapter {
+  private apiUrl: string;
+  private webhookId: string;
+  private token: string;
+  private regex: RegExp;
+  private httpClient: HttpAdapter;
+
+  constructor(params: Config) {
+    super(params);
+
+    this.apiUrl = 'discord.com/api';
+    this.webhookId = params.webhookId;
+    this.token = params.token;
+    this.regex = params.regex;
+    this.httpClient = new HttpAdapterFactory().create({});
+  }
+
+  public log(message: LogMessage): void {
+    const formattedMessageArray  = this.formatMessage(message);
+
+    if (formattedMessageArray.find((fmessage) => this.regex.test(fmessage))) {
+      this.publish(formattedMessageArray.join(' '));
+    }
+  }
+
+  private publish(message: string) {
+    try {
+      return this.httpClient.post<unknown>(`${this.apiUrl}/${this.webhookId}/${this.token}`, {
+        'content': message,
+      });
+    } catch (err) {
+      console.log('Something when wrong', err);
```

went*

---

Why don't let the error propagate to the appropriate error handler instead?